### PR TITLE
Spike a way of checking out alphagov github repos

### PR DESCRIPTION
**Pull request by jabley**

### CODE REVIEW

**Comment by KushalP** [comment]
@jabley What's the status of this?

**Comment by samjsharpe** [comment]
Sorry - hit the wrong button.

The status is that the code did work at one point. It's got a merge conflict now, but shouldn't be hard to fix. I was going to take some time to add the ability to be slightly more configurable (so I could use it instead of [what I do now](https://github.com/alphagov/gds-boxen/blob/master/modules/people/manifests/samjsharpe/repos.pp)), but I never found that time and I don't have any objection to what's here being merged (I can always extend it later).

**Comment by jabley** [comment]
Fixed merge issues.

File: modules/alphagov/manifests/init.pp [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+#           - spotlight
+#           - backdrop
+#
+class alphagov (
+  $repos   = [],
+) {
+
+  validate_array($repos)
+
+  if !(empty($repos)) {
```
Comment (jabley) [inline modules/alphagov/manifests/init.pp]: I've copied what `gds_ruby` does. If that's a bad example to copy, happy to change it.

**Comment by KushalP** [comment]
:+1:

**Pull request merged**